instrumentation/apache-httpasyncclient-4.1/src/test/java/io/opentelemetry/instrumentation/hypertrace/apachehttpasyncclient/ApacheAsyncClientInstrumentationModuleTest.java

```diff
@@ -16,138 +16,88 @@
-    Assertions.assertEquals(
-        "bar",
-        clientSpan.getAttributes().get(HypertraceSemanticAttributes.httpRequestHeader("foo")));
-    Assertions.assertNull(
-        clientSpan.getAttributes().get(HypertraceSemanticAttributes.HTTP_REQUEST_BODY));
-    SpanData responseBodySpan = traces.get(0).get(1);
-    Assertions.assertEquals(
-        GetJsonHandler.RESPONSE_BODY,
-        responseBodySpan.getAttributes().get(HypertraceSemanticAttributes.HTTP_RESPONSE_BODY));
+    Assertions.assertFalse(Span.current().isRecording());
+    return Pair.of(response.getStatusLine().getStatusCode(), responseBody);
   }
 
-  @Test
-  public void postJson()
-      throws IOException, TimeoutException, InterruptedException, ExecutionException {
-    StringEntity entity =
-        new StringEntity(JSON, ContentType.create(ContentType.APPLICATION_JSON.getMimeType()));
-    postJsonEntity(entity);
-  }
-
-  @Test
-  public void postJsonNonRepeatableEntity()
-      throws IOException, TimeoutException, InterruptedException, ExecutionException {
-    StringEntity entity = new NonRepeatableStringEntity(JSON);
-    postJsonEntity(entity);
-  }
-
-  public void postJsonEntity(HttpEntity entity)
-      throws TimeoutException, InterruptedException, IOException, ExecutionException {
-    HttpPost postRequest = new HttpPost();
-    postRequest.setEntity(entity);
-    postRequest.setHeader("Content-type", "application/json");
-    postRequest.setURI(
-        URI.create(String.format("http://localhost:%d/post", testHttpServer.port())));
-
-    Future<HttpResponse> responseFuture = client.execute(postRequest, new NoopFutureCallback());
-
+  @Override
+  public Pair<Integer, String> doGetRequest(String uri, Map<String, String> headers)
+      throws IOException, ExecutionException, InterruptedException {
+    HttpGet request = new HttpGet(uri);
+    for (String key : headers.keySet()) {
+      request.addHeader(key, headers.get(key));
+    }
+    Future<HttpResponse> responseFuture = client.execute(request, new NoopFutureCallback());
     HttpResponse response = responseFuture.get();
-    Assertions.assertEquals(204, response.getStatusLine().getStatusCode());
-
-    TEST_WRITER.waitForTraces(1);
-    List<List<SpanData>> traces = TEST_WRITER.getTraces();
-    Assertions.assertEquals(1, traces.size());
-    Assertions.assertEquals(1, traces.get(0).size());
-    SpanData clientSpan = traces.get(0).get(0);
-
-    String requestBody = readInputStream(entity.getContent());
-    Assertions.assertEquals(
-        "test-value",
-        clientSpan
-            .getAttributes()
-            .get(HypertraceSemanticAttributes.httpResponseHeader("test-response-header")));
-    Assertions.assertEquals(
-        requestBody,
-        clientSpan.getAttributes().get(HypertraceSemanticAttributes.HTTP_REQUEST_BODY));
-    Assertions.assertNull(
-        clientSpan.getAttributes().get(HypertraceSemanticAttributes.HTTP_RESPONSE_BODY));
+    Thread.sleep(200);
```

Comment:
The same as before, why do we need a static wait? We should avoid sleeps on a static time.

Reply:
The second span was not getting created without this wait. The test was flaky without this wait, and most times the second span with response body was not there.

Reply:
see the previous comment